ibis/backends/flink/__init__.py

```diff
@@ -341,44 +365,86 @@ def create_table(
         import pyarrow_hotfix  # noqa: F401
 
         import ibis.expr.types as ir
-        from ibis.backends.flink.datatypes import FlinkRowSchema
 
         if obj is None and schema is None:
-            raise exc.IbisError("The schema or obj parameter is required")
+            raise exc.IbisError("`schema` or `obj` is required")
+        if isinstance(obj, pd.DataFrame) and temp is False:
+            temp = True
```

Comment:
I agree that erroring is cleaner and more explicit than overriding, which is actually what I did at first :)
However with erroring, I encountered the following complication: `temp` is `False` by default for all backends and this makes sense to me.  In many places though, `create_table()` is called with in-memory `obj` and without `temp`, which triggers the error for `isinstance(obj, pd.DataFrame) and temp is False`. I could not find a clean way to modify the tests to make them fit this unusual behavior by Flink backend.

I switched to overriding with the following in mind: As discussed at some point, we might add support for creating tables (instead of creating views as we do now) which would enable supporting `temp=False` for in-memory `obj` as well. This would eliminate the need to override `temp`.

Given this background, do you have any other suggestion? Or should I just switch back to erroring (from overriding) and deal with the test failures by marking them for Flink?

Reply:
I would error and mark the tests as `notimpl`, since those tests are for creating non-temp tables.  Then if we figure out a good pattern for persistent table/view creation from in-memory tables, we'll get a nice signal that it's working.

Reply:
Thanks, switched to erroring.